core/domain/topic_domain.py

```diff
@@ -102,13 +127,94 @@ def to_dict(self):
         return topic_change_dict
 
 
+class Subtopic(object):
+    """Domain object for a Subtopic."""
+
+    def __init__(self, subtopic_id, title, skill_ids):
+        """Constructs a Subtopic domain object.
+
+        Args:
+            subtopic_id: str. The unique ID of the subtopic.
+            title: str. The title of the subtopic.
+            skill_ids: list(str). The list of skill ids that are part of this
+                subtopic.
+        """
+        self.id = subtopic_id
+        self.title = title
+        self.skill_ids = skill_ids
+
+    def to_dict(self):
+        """Returns a dict representing this Subtopic domain object.
+
+        Returns:
+            A dict, mapping all fields of Subtopic instance.
+        """
+        return {
+            'id': self.id,
+            'title': self.title,
+            'skill_ids': self.skill_ids
+        }
+
+    @classmethod
+    def from_dict(cls, subtopic_dict):
+        """Returns a Subtopic domain object from a dict.
+
+        Args:
+            subtopic_dict: dict. The dict representation of Subtopic object.
+
+        Returns:
+            Subtopic. The corresponding Subtopic domain object.
+        """
+        subtopic = cls(
+            subtopic_dict['id'], subtopic_dict['title'],
+            subtopic_dict['skill_ids'])
+        return subtopic
+
+    @classmethod
+    def create_default_subtopic(cls, subtopic_id, title):
+        """Creates a Subtopic object with default values.
+
+        Args:
+            subtopic_id: str. ID of the new subtopic.
+            title: str. The title for the new subtopic.
+
+        Returns:
+            Subtopic. A subtopic object with given id, title and empty skill ids
+                list.
+        """
+        return cls(subtopic_id, title, [])
+
+    def validate(self):
+        """Validates various properties of the Subtopic object.
+
+        Raises:
+            ValidationError: One or more attributes of the subtopic are
+                invalid.
+        """
+        if not isinstance(self.title, basestring):
+            raise utils.ValidationError(
+                'Expected subtopic title to be a string, received %s' %
+                self.title)
+        if not isinstance(self.skill_ids, list):
+            raise utils.ValidationError(
+                'Expected skill ids to be a list, received %s' %
```

Comment:
be more careful; need to also check each item inside it is a string

Reply:
Done.